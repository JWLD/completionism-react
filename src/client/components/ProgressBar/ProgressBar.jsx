import React from 'react';
import PropTypes from 'prop-types';

import content from 'constants/content';
import './ProgressBar.scss';

const ProgressBar = (props) => {
	const title = content[props.routeProps.content].xpac;

	const total = props.data.length;
	const collected = props.data.filter(item => props.storageData.includes(item.id)).length;
	const percentage = collected ? (collected / total) * 100 : 0;
	const progStyle = { width: `${percentage}%` };

	return (
		<div className="prog-bar-wrap">
			<div>
				<h2>{title}</h2>
				<span>{collected} / {total}</span>
			</div>

			<div className="prog-bar">
				<div style={progStyle} />
			</div>
		</div>
	);
};

ProgressBar.propTypes = {
	data: PropTypes.array.isRequired,
	routeProps: PropTypes.object.isRequired,
	storageData: PropTypes.array.isRequired
};

export default ProgressBar;
