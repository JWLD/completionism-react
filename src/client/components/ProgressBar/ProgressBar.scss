@import 'global';

.prog-bar-wrap {
	@include itemListWrap;
	flex-direction: column;

	> div:first-child {
		justify-content: space-between;
		width: 100%;

		h2, span {
			margin: -.5rem 0;
			font: 4rem DinMC;
			color: white;
			text-transform: uppercase;
		}
	}

	$progBarHeight: 1.5rem;

	.prog-bar {
		position: relative;
		overflow: hidden;
		height: $progBarHeight;
		width: 100%;
		margin-top: $itemListPadding;
		background-color: #333;
		border-radius: .3rem;

		div {
			position: absolute;
			top: 0;
			left: 0;
			height: $progBarHeight;
			width: 4rem;
			background-color: $pos;
			border-radius: .3rem;
		}
	}
}
