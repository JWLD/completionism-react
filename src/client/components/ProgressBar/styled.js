import styled from 'styled-components'

import { vars } from 'style'

export const ProgressBar = styled.div`
  position: relative;
  overflow: hidden;
  height: ${props => props.height}rem;
  min-height: ${props => props.height}rem;
  width: 100%;
  background-color: #444;
  border-radius: 0.3rem;
`

export const BarFill = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: ${props => props.height}rem;
  min-height: ${props => props.height}rem;
  width: ${props => props.fill}%;
  background-color: ${vars.pos};
  border-radius: 0.3rem;
`
