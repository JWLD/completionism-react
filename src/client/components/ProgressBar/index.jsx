import React from 'react'
import PropTypes from 'prop-types'

import * as SC from 'ProgressBar/styled'

const ProgressBar = ({ count, height }) => (
  <SC.ProgressBar height={height}>
    <SC.BarFill fill={count} height={height} />
  </SC.ProgressBar>
)

ProgressBar.propTypes = {
  count: PropTypes.number.isRequired,
  height: PropTypes.number
}

ProgressBar.defaultProps = {
  height: 1.5
}

export default ProgressBar
