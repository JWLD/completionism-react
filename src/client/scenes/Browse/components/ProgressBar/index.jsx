import React from 'react'
import PropTypes from 'prop-types'

import * as SC from './styled'

const ProgressBar = props => {
  const fillPercent = props.count / props.total * 100
  const fillProp = isNaN(fillPercent) ? 0 : fillPercent

  return (
    <SC.ProgressBarWrap>
      <SC.TextWrap>
        <SC.ProgressText>{props.title}</SC.ProgressText>
        <SC.ProgressText>
          {props.count} / {props.total}
        </SC.ProgressText>
      </SC.TextWrap>

      <SC.ProgressBar>
        <SC.BarFill fill={fillProp} />
      </SC.ProgressBar>
    </SC.ProgressBarWrap>
  )
}

ProgressBar.propTypes = {
  count: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  total: PropTypes.number.isRequired
}

export default ProgressBar
