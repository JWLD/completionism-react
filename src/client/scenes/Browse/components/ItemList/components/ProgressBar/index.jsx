import React from 'react'
import PropTypes from 'prop-types'

import * as SC from 'ProgressBar/styled'
import { BrowseBlock } from 'style/components'

const ProgressBar = props => {
  const fillPercent = props.count / props.total * 100
  const fillProp = isNaN(fillPercent) ? 0 : fillPercent

  return (
    <BrowseBlock>
      <SC.TextWrap>
        <SC.ProgressText>{props.title}</SC.ProgressText>
        <SC.ProgressText>
          {props.count} / {props.total}
        </SC.ProgressText>
      </SC.TextWrap>

      <SC.ProgressBar>
        <SC.BarFill fill={fillProp} />
      </SC.ProgressBar>
    </BrowseBlock>
  )
}

ProgressBar.propTypes = {
  count: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  total: PropTypes.number.isRequired
}

export default ProgressBar
