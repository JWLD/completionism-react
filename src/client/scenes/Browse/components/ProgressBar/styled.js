import styled from 'styled-components'

import { mixins, vars } from 'style'
import { BrowseBlock } from 'style/components'

const barHeight = 1.5

export const ProgressBarWrap = styled(BrowseBlock)`
  flex-direction: column;
`

export const TextWrap = styled.div`
  ${mixins.flex};

  justify-content: space-between;
  width: 100%;
`

export const ProgressText = styled.span`
  margin: -0.5rem 0;
  font: 3.5rem DinMC;
  color: white;
  text-transform: uppercase;
`

export const ProgressBar = styled.div`
  position: relative;
  overflow: hidden;
  height: ${barHeight}rem;
  width: 100%;
  margin-top: ${vars.itemListPadding}rem;
  background-color: #444;
  border-radius: 0.3rem;
`

export const BarFill = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: ${barHeight}rem;
  width: ${props => props.fill}%;
  background-color: ${vars.pos};
  border-radius: 0.3rem;
`
