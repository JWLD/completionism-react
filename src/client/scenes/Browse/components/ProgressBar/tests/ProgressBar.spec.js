import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import ProgressBar from 'ProgressBar'

Enzyme.configure({ adapter: new Adapter() })

let mockProps = {}

beforeEach(() => {
  mockProps = {
    count: 1,
    title: 'foo',
    total: 2
  }
})

describe('<ItemList />', () => {
  it('renders correctly', () => {
    const component = shallow(<ProgressBar {...mockProps} />)

    expect(component).toMatchSnapshot()
  })

  it('handles NaN percentage calculation correctly', () => {
    mockProps.count = 0
    mockProps.total = 0

    const component = shallow(<ProgressBar {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})
