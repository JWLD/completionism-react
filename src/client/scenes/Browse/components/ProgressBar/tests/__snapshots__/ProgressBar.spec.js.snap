// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemList /> handles NaN percentage calculation correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    count={0}
    title="foo"
    total={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TextWrap>
          <ProgressText>
            foo
          </ProgressText>
          <ProgressText>
            0
             / 
            0
          </ProgressText>
        </TextWrap>,
        <ProgressBar>
          <BarFill
            fill={0}
          />
        </ProgressBar>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ProgressText>
              foo
            </ProgressText>,
            <ProgressText>
              0
               / 
              0
            </ProgressText>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "foo",
            },
            "ref": null,
            "rendered": "foo",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                0,
                " / ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              0,
              " / ",
              0,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <BarFill
            fill={0}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fill": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TextWrap>
            <ProgressText>
              foo
            </ProgressText>
            <ProgressText>
              0
               / 
              0
            </ProgressText>
          </TextWrap>,
          <ProgressBar>
            <BarFill
              fill={0}
            />
          </ProgressBar>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ProgressText>
                foo
              </ProgressText>,
              <ProgressText>
                0
                 / 
                0
              </ProgressText>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "foo",
              },
              "ref": null,
              "rendered": "foo",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  0,
                  " / ",
                  0,
                ],
              },
              "ref": null,
              "rendered": Array [
                0,
                " / ",
                0,
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <BarFill
              fill={0}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fill": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ItemList /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    count={1}
    title="foo"
    total={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TextWrap>
          <ProgressText>
            foo
          </ProgressText>
          <ProgressText>
            1
             / 
            2
          </ProgressText>
        </TextWrap>,
        <ProgressBar>
          <BarFill
            fill={50}
          />
        </ProgressBar>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ProgressText>
              foo
            </ProgressText>,
            <ProgressText>
              1
               / 
              2
            </ProgressText>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "foo",
            },
            "ref": null,
            "rendered": "foo",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                1,
                " / ",
                2,
              ],
            },
            "ref": null,
            "rendered": Array [
              1,
              " / ",
              2,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <BarFill
            fill={50}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fill": 50,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TextWrap>
            <ProgressText>
              foo
            </ProgressText>
            <ProgressText>
              1
               / 
              2
            </ProgressText>
          </TextWrap>,
          <ProgressBar>
            <BarFill
              fill={50}
            />
          </ProgressBar>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ProgressText>
                foo
              </ProgressText>,
              <ProgressText>
                1
                 / 
                2
              </ProgressText>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "foo",
              },
              "ref": null,
              "rendered": "foo",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  1,
                  " / ",
                  2,
                ],
              },
              "ref": null,
              "rendered": Array [
                1,
                " / ",
                2,
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <BarFill
              fill={50}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fill": 50,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
